packages/osd-std/src/json.ts

```diff
@@ -0,0 +1,321 @@
+    : (markedText: string) => JSON.parse(markedText, (key, val) => convertMarkedValues(val));
+
+  /* RegExp cannot replace AST and the process of marking adds quotes. So, any false-positive hit
+   * will make the JSON string unparseable.
+   *
+   * To find those instances, we try to parse and watch for the location of any errors. If an error
+   * is caused by the marking, we remove that single marking and try again.
+   */
+  do {
+    try {
+      hadException = false;
+      obj = parseMarkedText(markedJSON);
+    } catch (e) {
+      hadException = true;
+      /* There are two types of exception objects that can be raised:
+       *  1) a proper object with lineNumber and columnNumber which we can use
+       *  2) a textual message with the position that we need to parse
+       */
+      let { lineNumber, columnNumber } = e;
+      if (!lineNumber || !columnNumber) {
+        const match = e?.message?.match?.(/^Unexpected token.*at position (\d+)$/);
+        if (match) {
+          lineNumber = 1;
+          // The position is zero-indexed; adding 1 to normalize it for the -2 that comes later
+          columnNumber = parseInt(match[1], 10) + 1;
+        }
+      }
+
+      if (lineNumber < 1 || columnNumber < 2) {
+        /* The problem is not with this replacement.
+         * Note: This will never happen because the outer parse would have already thrown.
+         */
+        // coverage:ignore-line
+        throw e;
+      }
+
+      /* We need to skip e.lineNumber - 1 number of `\n` occurrences.
+       * Then, we need to go to e.columnNumber - 2 to look for `"<mark>\d+"`; we need to `-1` to
+       * account for the quote but an additional `-1` is needed because columnNumber starts from 1.
+       */
+      const re = new RegExp(
+        `^((?:.*\\n){${lineNumber - 1}}[^\\n]{${columnNumber - 2}})"${marker}(-?\\d+)"`
+      );
+      if (!re.test(markedJSON)) {
+        /* The exception is not caused by adding the marker.
+         * Note: This will never happen because the outer parse would have already thrown.
+         */
+        // coverage:ignore-line
+        throw e;
+      }
+
+      // We have found a bad replacement; let's remove it.
+      markedJSON = markedJSON.replace(re, '$1$2');
+    }
+  } while (hadException);
+
+  return obj;
+};
+
+const stringifyObjectWithBigInts = (
+  obj: any,
+  candidate: string,
+  replacer?: ((this: any, key: string, value: any) => any) | null,
+  space?: string | number
+): string => {
+  const { marker } = getMarker(candidate);
+
+  /* The matcher that looks for "<marker><numerals>"
+   * Because we have made sure that `marker` was never present in the original object, we can
+   * carelessly assume every "<marker><numerals>" is due to our marking.
+   */
+  const markedBigIntMatcher = new RegExp(`"${marker}(-?\\d+)"`, 'g');
+
+  /* Convert BigInt values to a string and mark them.
+   * Can't be bothered with Number values outside the safe range because they are already corrupted.
```

Comment:
Do we want to throw some sort of error in that case?

Reply:
Those long ones would have already lost precision so letting them pass through will not ruin anything.